test/test_jit_autocast.py

```diff
@@ -524,10 +524,10 @@ def t_cpu_amp_autocast(x, y):
 
         x = torch.randn(5, 5, device="cuda", dtype=torch.float32)
         y = torch.randn(5, 5, device="cuda", dtype=torch.float32)
-        self._test_autocast(t_autocast_cpu, "aten::_autocast_to_reduced_precision", x, y)
+        # self._test_autocast(t_autocast_cpu, "aten::_autocast_to_reduced_precision", x, y)
         self._test_autocast(t_autocast_cuda, "aten::_autocast_to_reduced_precision", x, y)
         self._test_autocast(t_cuda_amp_autocast, "aten::_autocast_to_reduced_precision", x, y)
-        self._test_autocast(t_cpu_amp_autocast, "aten::_autocast_to_reduced_precision", x, y)
+        # self._test_autocast(t_cpu_amp_autocast, "aten::_autocast_to_reduced_precision", x, y)
```

Comment:
Hey Ed I was going through PR failures (finally trying to land this PR again), and making autocast a per-backend functionality key may be harder than we thought.

Right now, the user API for autocast is something like:
```
with torch.cuda.amp.autocast()
    # only the operations on CUDA tensors will flow through AMP

with torch.cpu.amp.autocast()
    # only the operations on CPU tensors will flow through AMP
```

And there's also `torch.jit.scripting` capture logic to keep this consistent when you script.

If "turning on AMP for a single backend at a time" is really functionality that we want to support, then that won't fit too well into the "per-backend functionality key" model, because we now only have one functionality bit for autocast, that you can add/remove to TLS (`AutocastFunctionality`).

If we don't actually care about supporting that use case though, do you think it's worth trying to change the autocast API (and dealing with the BC breakage?)

If not, I'll just remove the autocast bits from this PR, and land the rest - fixing up the dispatch key printing.